src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/DynamicTests.cs

```diff
@@ -643,6 +643,238 @@ .maxstack  9
 ");
         }
 
+        [WorkItem(1160855, "DevDiv")]
+        [Fact]
+        public void AwaitDynamic()
+        {
+            var source = @"
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+class C
+{
+    dynamic d;
+
+    void M(int p)
+    {
+        d.Test(); // Force reference to runtime binder.
+    }
+
+    static void F(Func<object> f)
+    {
+    }
+
+    static void G(Func<Task<object>> f)
+    {
+    }
+}
+";
+            var comp = CreateCompilationWithMscorlib45(source, new[] { SystemCoreRef, CSharpRef }, TestOptions.DebugDll);
+            var runtime = CreateRuntimeInstance(comp);
+            var context = CreateMethodContext(runtime, "C.M");
+            var testData = new CompilationTestData();
+            string error;
+            var result = context.CompileExpression("G(async() => await d())", out error, testData);
+            Assert.Null(error);
+            VerifyCustomTypeInfo(result, null);
+            var methodData = testData.GetMethodData("<>x.<>c__DisplayClass0_0.<<<>m0>b__0>d.System.Runtime.CompilerServices.IAsyncStateMachine.MoveNext()");
+            methodData.VerifyIL(@"
```

Comment:
It worries me a little bit to baseline this much IL (seems fragile), but I'm not sure what I'd suggest as an alternative besides a SubString match...

Probably not worth worrying about until it becomes a problem, but thought I'd mention it in case anyone has ideas...